w06h01/test/pgdp/LoopTest.java

```diff
@@ -18,39 +21,53 @@ public class LoopTest {
     @Test
-    public void testUsageOfLoops() {
-        String[] filePaths = new String[] {
-                "src/pgdp/datastructures/lists/RecIntList.java",
-                "src/pgdp/datastructures/lists/RecIntListElement.java"
-        };
-
-        String[] notAllowedRegExp = new String[] {
+    public void testUsageOfLoops() throws IOException {
+        String[] notAllowedRegExp = new String[]{
                 "for\s*[(]{1}",
                 "while\s*[(]{1}",
                 "Stream.of\\(",
                 "Stream<",
                 "Arrays.stream\\(",
         };
 
-        for (String path : filePaths) {
-            // Methods that are allowed to use loops
-            String[] allowedMethods = new String[] {
-                    "public static void main(String[] args) {",
-                    "public String toString() {",
-                    "public String toConnectionString() {"
-            };
-
-            String file = readFile(path);
-            String filteredFile = removeMultipleMethods(file, allowedMethods);
-
-            // Check if there are any loops
-            for (String regExp : notAllowedRegExp) {
-                Pattern pattern = Pattern.compile(regExp);
-                Matcher matcher = pattern.matcher(filteredFile);
-                assertFalse(matcher.find(),
-                        "You are not allowed to use loops in your solution!");
+        try (Stream<Path> sourceFiles = Files.find(Path.of("src"), 100, (filePath, fileAttr) -> fileAttr.isRegularFile() && filePath.toString().endsWith(".java"))) {
+            sourceFiles.forEach(f -> {
+                String path = f.toString();
+
+
+                // Methods that are allowed to use loops
+                String[] allowedMethods = new String[]{
+                        "public static void main(String[] args) {",
+                        "public String toString() {",
+                        "public String toConnectionString() {"
+                };
+
+                String file = readFile(path);
+                String filteredFile = removeMultipleMethods(file, allowedMethods);
+
+                // Check if there are any loops
+                for (String regExp : notAllowedRegExp) {
+                    int lineNum = getOffendingLineNum(filteredFile, regExp);
+                    boolean noLoop = lineNum == -1;
+
+                    assertTrue(noLoop,
+                            "You are not allowed to use loops in your solution! Found a loop at " + path + ":" + lineNum);
+                }
+            });
+        }
+    }
+
+    private int getOffendingLineNum(String s, String regExp) {
+        Pattern pattern = Pattern.compile(regExp);
+
+        String[] lines = s.split("\n");
+        for (int i = 0; i < lines.length; i++) {
+            Matcher matcher = pattern.matcher(lines[i]);
+            if (matcher.find()) {
+                return i + 1;
             }
         }
+        return -1;
```

Comment:
Just a small comment to indicate what `-1` means

```suggestion
        // No loop found, returning -1
        return -1;
```